src/containers/ToDoList/ToDoList.jsx

```diff
@@ -54,6 +58,23 @@ export const ToDoList = () => {
     }
   };
 
+  const handleClearCompletedTasks = () => {
+    setTasks(tasks.filter((task) => !task.checked));
+  };
+
+  const handleFilterTask = () => {
+    switch (filterTask) {
+      case "all":
+        return tasks;
+      case "active":
+        return tasks.filter((task) => !task.checked);
+      case "completed":
+        return tasks.filter((task) => task.checked);
+      default:
+        return tasks;
+    }
+  };
```

Comment:
1. This doesn't have to be a function
2. Switches are not wrong, they do their job, but still I'd personally wanna have this rewritten to something of more like:

```
const taskFilters = {
  active: (task) => !task.checked, // a function can be a value in JSON object too!
  completed: (task) => task.checked
}

const filteredTasks = activeFilter ? tasks.filter(filters[activeFilter]) : tasks
```
This assumes, that if `activeFilter` is `undefined`, we don't apply any filter = display all. So state `all` won't be needed here.

See how many lines more you'll have to remove :-)
Also let me know, if this approach makes sense to you.

Reply:
it's definitely more readable, but I'm not sure if I implemented it correctly

Reply:
You did well :-)